lib/rubocop/cop/style/redundant_map_compact.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for usage of `select` or `reject`
```

Comment:
When a block has `if` condition, it cannot be replaced with `reject`.

```suggestion
      # This cop checks for usage of `select`
```

Instead, it could add text to the description, such as "replace with `reject` when the block has `unless` condition".

Reply:
I think such an example using `if` can be rewritten as `reject`.

```ruby
# bad
[1, 2, 3, 4].map do |e|
  next if e.odd?

  e
end.compact
# => [2, 4]

# good
[1, 2, 3, 4].reject { |e| e.odd? }
# => [2, 4]
```

Reply:
The "instead of `map { |value| simple_condition? ? value : next }.compact`." at L7 could be bad.
Would modifying it as "instead of `map { ... }.compact`." solve the problem?